airflow/operators/datetime_branch.py

```diff
@@ -0,0 +1,101 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import datetime
+from typing import Dict, Iterable, Union
+
+from airflow.exceptions import AirflowException
+from airflow.operators.branch_operator import BaseBranchOperator
+from airflow.utils import timezone
+from airflow.utils.decorators import apply_defaults
+
+
+class DateTimeBranchOperator(BaseBranchOperator):
+    """
+    Branches into one of two lists of tasks depending on the current datetime.
+
+    True branch will be returned when `datetime.datetime.now()` falls below
+    `target_upper` and above `target_lower`.
```

Comment:
What's the behaviour when backfilling or catching-up ?

Reply:
How about adding something like:

https://github.com/apache/airflow/blob/e363ac1f33cd82188fb0923791f5cb747629d501/airflow/operators/weekday.py#L86-L89

cc @eladkal

Reply:
Good call! Behaviour when backfilling would be to still use system date, which is probably not what a user would expect. I added a `use_task_execution_date` argument very similar to the one you linked, with some minor adaptation since we are talking about dates instead of days of the week.